### PR TITLE
    overlay/live: Support "fulliso" path as well

### PR DESCRIPTION
**Pull request by cgwalters**
Supports https://github.com/coreos/coreos-assembler/pull/928
   
Here `fulliso` images have a few changes; what this code
is concerned with is finding the squashfs in the ISO, not the initrd.
    
We continue to support `live` images as they existed before
as well.

### CODE REVIEW

**Comment by cgwalters** [comment]
I'm going to rework this to support both `live` and `fulliso` cases.

**Comment by cgwalters** [comment]
Now tested in concert with https://github.com/coreos/coreos-assembler/pull/928 !

**Comment by jlebon** [comment]
We can un-hold this one now, right?

**Comment by cgwalters** [comment]
Yep, un-held.

**Review by jlebon — APPROVED** [review]
OK I think I understand this better now.

One bikeshed is around naming (should've brought this up in https://github.com/coreos/coreos-assembler/pull/928). Both the `live` and `fulliso` builds are "live", right? Feels like `fulliso` should reflect that too. WDYT about e.g. `fulliso-live`?

Code itself LGTM!

**Comment by cgwalters** [comment]
Maybe `liveinst`?

**Comment by jlebon** [comment]
> Maybe `liveinst`?

Hmm, or `livemedia`? To reflect the fact you need the actual media available.

**Comment by cgwalters** [comment]
In that it doesn't do PXE?  Hrmm.  I think the `inst` is more descriptive in that it's telling you it's for installing, whereas `live` *can* be for that but is also intended for "diskless PXE" type scenarios.

One tangentially related thing here is I'm considering reworking things so that we just boot off of the metal image directly.  This would mean we'd need to decompress the image all at once (making booting a lot slower) but would save a lot of storage/network space.  And it'd more strongly orient the use case towards installing.

**Comment by cgwalters** [comment]
I also want to investigate whether we can do compression at with something like [VDO](https://www.redhat.com/de/blog/look-vdo-new-linux-compression-layer) which would be pretty ideal for this use case.

**Comment by jlebon** [comment]
> In that it doesn't do PXE? Hrmm. I think the `inst` is more descriptive in that it's telling you it's for installing, whereas `live` can be for that but is also intended for "diskless PXE" type scenarios.

Sure, WFM!

**Comment by dustymabe** [comment]
I have to admit I missed the other two PRs regarding fulliso. I think it'd be worth discussing our strategy here regarding what ISOs we provide. It would be nice if we didn't provide multiple ISOs (ironically I've been doing work this week to drop the old installer ISOs so that we could just ship the live ISOs), but I realize there are different use cases that have different requirements. Maybe we can discuss soon.

**Comment by cgwalters** [comment]
FWIW, I was intending fulliso for FCOSB (Fedora Silverblue-based-on-FCOS), not necessarily FCOS.  Having the build capability doesn't mean we need to ship it for FCOS.

**Comment by dustymabe** [comment]
That helps. I'd still like to chat sometime about it just to bring my understanding up to par, if you don't mind. Note: I'm not blocking or intending to block this PR.

**Comment by cgwalters** [comment]
OK I think we can do a rename/changes in a second pass.  I'd like to prove out the functionality first.

File: overlay.d/05core/usr/lib/dracut/modules.d/20live/live-generator [hunk 1]
```diff
@@ -13,7 +15,8 @@ add_requires() {
     ln -sf "../${name}" "${requires_dir}/${name}"
 }
 
-if [ -e /root.squashfs ]; then
+isoroot=$(getarg coreos.liveiso=)
```
Comment (SerialVelocity) [inline overlay.d/05core/usr/lib/dracut/modules.d/20live/live-generator:18]: @cgwalters I think this is now causing the generator to fail if this isn't specified:
```
/usr/lib/systemd/system-generators/live-generator failed with exit status 1
```

Comment (cgwalters) [reply overlay.d/05core/usr/lib/dracut/modules.d/20live/live-generator:18]: Thanks for the report!  https://github.com/coreos/fedora-coreos-config/pull/260

**Pull request merged**